Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -7985,37 +7985,29 @@ def __init__(self,
         self._size = size
         self._to_date = to_date
         self._value = value
-        self._original_limit = limit
-        self._next_limit = limit
-        self._search_is_done = False
+        self._limit = limit
 
     def __iter__(self):
-        self._total = None
-        self._search_after_param = None
-        self._page = self._original_page
-        self._next_limit = self._original_limit
-        self._search_is_done = False
         return self
 
     # python2
     def next(self):
         return self.__next__()
 
     def __next__(self):
-        if self._search_is_done:
+        if self.is_search_done():
             raise StopIteration
-        size = min(self._size, self._next_limit or self._size)
         res = self.search_indicators_by_version(from_date=self._from_date,
                                                 query=self._query,
-                                                size=size,
+                                                size=self._size,
                                                 to_date=self._to_date,
                                                 value=self._value)
         fetched_len = len(res.get('iocs') or [])
         if fetched_len == 0:
+            self.limit = 0
```

Comment:
replace limit with count - i.e. instead of lowering the limit, we'll increase the count.